### DISCUSSION TITLE
container/kube-scheduler

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#212] [General]
# container/kube-scheduler

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/container/kube-scheduler.html

**Comment by shihaitao2005** [comment]
例如：JobA 需要 4 个 Pod 同时启动，才算正常运行。kube-scheduler 依次调度 3 个 Pod 并创建成功，到第 4 个 Pod 时，集群资源不足，则 JobA 的 3 个 Pod 处于空等的状态。但是它们已经占用了部分资源，如果第 4 个 Pod 不能及时启动的话，整个 JobA 无法成功运行，更糟糕的集群其他的资源刚好被 JobB 的 3 个 Pod 所占用，同时在等待 JobB 的第 4 个 Pod 创建，此时整个集群就出现了死锁。----JobB 应该是JobA

**Comment by zjyyzhengli** [comment]
关于图片编号的几个问题：
1. “**图 7-1** 中的 Scheduler Cache” -- 这个图片编号有误，似乎是“图 7-36”？
2. "根据**图 7-37** 可以看出，Kubernetes 调度的核心在于两个互相独立的控制循环。" -- 这里的图片编号根据上下文意思，应该是**图7-36**
3. “触发两个最核心的调度阶段：预选阶段（**图 7-31** 中的 Predicates）和优选阶段（**图 7-31** 中的 Priority）” -- 这里的图片编号应该也是**图7-36**

**Reply by isno** [reply]
感谢，已修改
